### DISCUSSION TITLE
Formulas/Calculated fields?

### DISCUSSION THREAD

**Discussion by arladmin** [Budibase/budibase#1033] [Help] [answered]
Hi.

Is it possible to implement formula based fields? Formula could be stand-alone or based on other fields (e.g. Airtable Formulas + Roll Up/Look Up options)?
If not, is it on the roadmap already?

Thanks.

**Comment by mjashanks** [comment]
Hey @arladmin. This is not possible with the current release - 0.6.2.

This is on our roadmap and will come in stages.

1. PR #1020 will allow some manipulation of data in the client app. This would be most commonly used for formatting data. E.g. format a date as "Thursday, January 28th". However, this could also be used to do some basic calculations. Also, see here: https://github.com/Budibase/budibase/blob/feature/handlebars-helpers/packages/string-templates/README.md. This will be part of the next release.

2. Adding a formula column type. This is closer to what you are asking. We hope to implement this over the next few months, although we do not currently have a solid plan. This would allow you to create a formula based on other fields in the same table. Formulas will most likely be written in JavaScript.

3. Rollup formulas. We can very much see the benefit of this, but will likely be a bit further out. No immediate plans to implement this, but I can see that it would be a popular request.

Feel free to ask more questions or share your opinion!

**Reply by arladmin** [reply]
@mjashanks Thanks for your answer. 
Eagerly looking forward to these new features!

**Reply by hendrikjap** [reply]
@mjashanks , looks forward to Rollup formula!

**Reply by ronaldbalza** [reply]
Any update on this, please?

**Reply by shogunpurple** [reply]
Hi @ronaldbalza - this feature has been shipped and is available in the later versions of budibase now. Here's the documentation:

https://docs.budibase.com/building-apps/data/data-types/formula

**Reply by ronaldbalza** [reply]
Thanks @shogunpurple! Can't wait to test this awesome app builder :) Cheers!

**Comment by shogunpurple** [comment, marked as answer]
Looks like this was an older discussion. This feature has been shipped and available for some time now:

https://docs.budibase.com/building-apps/data/data-types/formula
